packages/pigment-react/tests/styled/styled.test.ts

```diff
@@ -0,0 +1,44 @@
+import path from 'node:path';
+import { expect } from 'chai';
+import { runTransformation } from '../testUtils';
+
+const theme = {
+  palette: {
+    primary: {
+      main: 'red',
+    },
+  },
+  size: {
+    font: {
+      h1: '3rem',
+    },
+  },
+  components: {
+    MuiSlider: {
+      styleOverrides: {
+        rail: {
+          fontSize: '3rem',
+        },
+      },
+    },
+  },
+};
+
+const CUSTOM_ERROR =
+  'The file contents have changed. Run "test:update" command to update the file if this is expected.';
+
+describe('Pigment CSS - styled', () => {
```

Comment:
All the test files are repetitive in code. Only thing that is changing is the file path.
Creating new tests means copying the same content everytime. This was already solved in the previous implementation.

Reply:
Added a comment above that'll handle duplication while still keeping the tests independent.

Reply:
> Creating new tests means copying the same content everytime. This was already solved in the previous implementation.

I'd argue this. The previous implementation does not let me use the test infra APIs like `it.only()` to run a specific test. Also, a specific theme for each test is not possible.

This made me think of [The wrong abstraction](https://sandimetz.com/blog/2016/1/20/the-wrong-abstraction). The agree that there are duplications but the tests should also be flexible to add/remove/alter as well.

cc @mnajdova for 3rd opinion.